server/utils/utils.ts

```diff
@@ -90,13 +90,16 @@ export const initialiseName = (fullName?: string): string | null => {
 export const hasValueChanged = (oldValue, newValue) => {
   if (!oldValue && newValue === '') return false
 
-  const newVal = newValue?.toUpperCase().trim()
-  const oldVal = oldValue?.toUpperCase().trim()
+  const newVal = newValue?.toUpperCase().trim() || undefined
+  const oldVal = oldValue?.toUpperCase().trim() || undefined
   return newVal !== oldVal
 }
 
-export const trimAllValuesInObjectArray = arr => {
-  return arr?.filter(obj => obj.name !== '').map(obj => ({ name: obj.name?.trim() }))
```

Comment:
refactor: if there is an array objects, exclude any that are empty strings (as would be passed in via req.body) and trim the rest